core/base/src/TApplication.cxx

```diff
@@ -628,20 +630,453 @@ TApplication::EExitOnException TApplication::ExitOnException(TApplication::EExit
-/// Print help on interpreter.
+/// The function generates a URL address for class or namespace (name).
+/// This is the URL to the online reference guide, generated by Doxygen.
+/// With the enumeration "EUrl" we pick which case we need - the one for
+/// class (kURLforClass) or the one for namespace (kURLforNameSpace).
+///
+/// \param[in] name <the name of the class or the namespace
+/// \param[in] type <the enumerator for class or namespace
 
-void TApplication::Help(const char *line)
+static TString UrlGenerator(const TString &name, EUrl type)
 {
-   gInterpreter->ProcessLine(line);
+   TString url("");
+   if (type == kURLforClass){
+      url.Append("https://root.cern.ch/doc/master/class");
+   } else if (type == kURLforNameSpace) {
+      url.Append("https://root.cern/doc/master/namespace");
+   }
+   url.Append(name);
+   url.Append(".html");
+   return url;
+}
+}
+
+namespace {
+////////////////////////////////////////////////////////////////////////////////
+/// The function returns a TString with the arguments of a member function
+/// (methodName) of a class (className), but modified with respect to Doxygen
+/// syntax - spacing around special sumbols and adding the missing sumbols
+/// ("std::", ",..."). "FormatMethodForDoxygen" works for the functions defined
+/// inside namespaces as well. We pass pointer to the class as well, to make
+/// sure the class exists.
+///
+/// \param[in] className <the name of the class
+/// \param[in] classVariable <pointer to the class
+/// \param[in] methodName <the name of the method from the class
 
-   Printf("\nROOT special commands.");
-   Printf("===========================================================================");
-   Printf("             pwd          : show current directory, pad and style");
-   Printf("             ls           : list contents of current directory");
-   Printf("             which [file] : shows path of macro file");
+static TString FormatMethodForDoxygen(const TString &className, TClass *classVariable, const TString &methodName)
+{
+   // With "GetSignature" we get the arguments of "methodName" and put them in a TString.
+   TString methodArguments(((TFunction *)(classVariable->GetListOfMethods()->FindObject(methodName)))->GetSignature());
+   // The TString "MethodArguments" is modified with respect of Doxygen requirements.
+   methodArguments.ReplaceAll(" = ", "=");
+   // maybe add also " =" and "= ";
+   methodArguments.ReplaceAll("* ", " *");
+   // is this always? below
+   methodArguments.ReplaceAll("*=", " *=");
+   methodArguments.ReplaceAll("*)", " *)");
+   methodArguments.ReplaceAll("*,", " *,");
+   methodArguments.ReplaceAll("*& ", " *&");
+   methodArguments.ReplaceAll("& ", " &");
+   methodArguments.ReplaceAll("ostream", "std::ostream");
+   methodArguments.ReplaceAll("istream", "std::istream");
+   methodArguments.ReplaceAll("*msgfmt", "*msgfmt,...");
+   // We need to take the regular expresion of the arguments (mostly if enumeration appears).
```

Comment:
Protect this replacement block by `if (classVariable) {` - we don't need to do that for functions in the global scope.